From 6f2ad70131d5dbf820377b9a193e7789a280ec08 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:05 -0400
Subject: [ethernet] i40evf: clean up memsets

Message-id: <1409658042-22689-5-git-send-email-sassmann@redhat.com>
Patchwork-id: 90198
O-Subject: [RHEL7.1 PATCH 004/281] i40evf: clean up memsets
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

As politely pointed out by Dave Miller, calls to memset do not need a
void pointer cast. Additionally, it is preferred to use sizeof(*the
actual object) instead of sizeof(type).

Change-ID: Id6a02429b7040111531f3865ea03fbe619167cb3
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Kevin Scott <kevin.c.scott@intel.com>
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit a63fa1cdb667609cd9f99435ecbbcda61861b38f)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_adminq.c b/drivers/net/ethernet/intel/i40e/i40e_adminq.c
index a50e6b3..ed3902b 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_adminq.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_adminq.c
@@ -647,9 +647,8 @@ static u16 i40e_clean_asq(struct i40e_hw *hw)
    desc_cb = *desc;
    cb_func(hw, &desc_cb);
   }
-  memset((void *)desc, 0, sizeof(struct i40e_aq_desc));
-  memset((void *)details, 0,
-         sizeof(struct i40e_asq_cmd_details));
+  memset(desc, 0, sizeof(*desc));
+  memset(details, 0, sizeof(*details));
   ntc++;
   if (ntc == asq->count)
    ntc = 0;
-- 
1.7.1